spec/system/visit_msa_component_new_spec.rb

```diff
@@ -40,6 +41,22 @@
     end
   end
 
+  context 'creation fails without unique entity id' do
+    it 'when entity id already exists for a msa component' do
+      existing_component = msa_component
+      entity = 'existing entity_id'
```

Comment:
Wouldn't this fail now due to your new validation to not have spaces in entity ids?